### PR TITLE
Refactor into unit tests and an e2e test

### LINKED ISSUE #16: Refactor into unit tests
https://github.com/r55-eth/r55/blob/main/r55/src/main.rs#L121-L197 these should become unit tests

### PR DESCRIPTION
**Pull request by fakedev9999**
Resolves #16.

Refactor existing tests into unit tests and an e2e test.
* Now, there's r55 lib instead of binary
* Separated test util functions from lib.rs
* Implement `compile_with_prefix`
* Implement `get_selector_from_sig`
* Change prints into assertions
* Use logger instead of prints

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -136,8 +136,8 @@ second transaction that indeed the balance is 42 (0x2a).
 # Architecture
 
 The compiler uses `rustc`, `llvm`,
-[eth-riscv-syscalls](https://github.com/leonardoalt/r55/tree/main/eth-riscv-syscalls),
-[eth-riscv-runtime](https://github.com/leonardoalt/r55/tree/main/eth-riscv-runtime)
+[eth-riscv-syscalls](https://github.com/r55-eth/r55/tree/main/eth-riscv-syscalls),
+[eth-riscv-runtime](https://github.com/r55-eth/r55/tree/main/eth-riscv-runtime)
 and [riscv-rt](https://github.com/rust-embedded/riscv/tree/master/riscv-rt) to
 compile and link ELF binaries with low-level syscalls to be executed by
 [rvemu-r55](https://github.com/lvella/rvemu):
```
Comment (leonardoalt) [inline README.md:143]: We should also update this one to r55 org

File: r55/tests/e2e.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+#[test]
+fn erc20() {
+    const ERC20_PATH: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/../erc20");
+    let rv_bytecode = compile_deploy(ERC20_PATH).unwrap();
+    let mut db = InMemoryDB::default();
+
+    let mut bytecode = vec![0xff];
+    bytecode.extend_from_slice(&rv_bytecode);
+
```
Comment (leonardoalt) [inline r55/tests/e2e.rs:20]: we could extract a helper that does the compilation and adds the `0xff` prefix

File: r55/tests/e2e.rs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    let addr = deploy_contract(&mut db, Bytes::from(bytecode)).unwrap();
+
+    let selector_balance = &keccak256("balance_of")[0..4];
```
Comment (leonardoalt) [inline r55/tests/e2e.rs:23]: this could get a helper that gets the sig from string

File: r55/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+    if let Err(e) = file.read_to_end(&mut bytecode) {
+        eyre::bail!("Failed to read file: {}", e);
+    }
+
+    Ok(bytecode)
+}
+
+pub fn compile_deploy(path: &str) -> eyre::Result<Vec<u8>> {
+    compile_runtime(path)?;
+    println!("Compiling deploy: {}", path);
```
Comment (leonardoalt) [inline r55/src/lib.rs:57]: lets set up a logger and use log::info instead of println whenever feasible

File: r55/tests/e2e.rs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    exec::{deploy_contract, run_tx},
+    test_utils::add_balance_to_db,
+};
+use revm::{
+    primitives::{address, keccak256, Address, Bytes},
+    InMemoryDB,
+};
+
+#[test]
+fn erc20() {
```
Comment (leonardoalt) [inline r55/tests/e2e.rs:13]: What's the difference between this test and the one in `lib.rs`?

File: README.md (hunk 1 above)
Comment (fakedev9999) [reply README.md]: Thanks for letting me know!

File: r55/tests/e2e.rs (hunk 2 above)
Comment (fakedev9999) [reply r55/tests/e2e.rs]: Agree. Implemented a new method `compile_with_prefix`.

File: r55/src/lib.rs (hunk 4 above)
Comment (fakedev9999) [reply r55/src/lib.rs]: Agree. Implemented a new method `initialize_logger` and set up logger.

File: r55/tests/e2e.rs (hunk 5 above)
Comment (leonardoalt) [reply r55/tests/e2e.rs:15]: makes sense!

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**